chromium_src/components/password_manager/core/browser/ui/weak_check_utility.cc

```diff
@@ -0,0 +1,22 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "src/components/password_manager/core/browser/ui/weak_check_utility.cc"
+
+#include "base/strings/utf_string_conversions.h"
+
+namespace password_manager {
+
+int GetPasswordStrength(const std::string& password) {
+  if (password.empty()) {
+    return 0;
```

Comment:
Is this particular check something we should upstream?

Reply:
Honestly, we should upstream the entire thing, but mills of the gods have been grinding extremely slowly in upstream land... I did touch on this with @bridiver, too — will try to see if they're open to it.

Reply:
Is there a good reason why we are using `0` as a sentinel for empty password and diverging from what upstream is doing?

Reply:
An empty password is very weak indeed :)

Reply:
I think a better question is: why does upstream not differentiate between these scenarios? An empty password is not weak, as its strength is non-existent. https://lowe.github.io/tryzxcvbn backs this up quite well — you can't get the strength of the empty string.

Reply:
I see, so the assumption is that a precondition for a password is that the string is not empty. That makes sense. I brought this up because I feel if we could upstream these changes, we potentially should, because then we shift the maintenance burden to them. It would also make things less contextual to align our approach with theirs.

Reply:
Agreed, I'll try to talk to the owners of the `Password Manager` to see what they think — no promises, though, as it's become quite cumbersome to work with upstream folks recently. I'm also filing 2 bugs for them for the "known issues" in the PR description.

Reply:
Pinged 5 people.